Tchap/Modules/Settings/SettingsViewController.m

```diff
@@ -2661,4 +2708,34 @@ - (void)changePasswordAlertPresenterDidTapChangePasswordAction:(ChangePasswordAl
 
 #endif
 
+#pragma mark - MXKDocumentPickerPresenterDelegate
+
+- (void)documentPickerPresenterWasCancelled:(MXKDocumentPickerPresenter *)presenter
+{
+    self.documentPickerPresenter = nil;
+}
+
+- (void)documentPickerPresenter:(MXKDocumentPickerPresenter *)presenter didPickDocumentsAt:(NSURL *)url
+{
+    self.documentPickerPresenter = nil;
+    
+    if ([MXMegolmExportEncryption isMegolmKeyFile:url])
+    {
+        // Show the keys import dialog
+        self->importView = [[MXKEncryptionKeysImportView alloc] initWithMatrixSession:self.mainSession];
+        self->currentAlert = self->importView.alertController;
+        // WARNING: SettingsViewController should extends MXKViewController
+        // It's working because SettingsViewController has startActivityIndicator and
+        // stopActivityIndicator methods
+        [self->importView showInViewController:self toImportKeys:url onComplete:^{
+            self->currentAlert = nil;
+            [[AppDelegate theDelegate] showAlertWithTitle: NSLocalizedStringFromTable(@"settings_crypto_import", @"Tchap", nil) message: NSLocalizedStringFromTable(@"settings_crypto_import_success", @"Tchap", nil)];
```

Comment:
There is a bug here when the user fills the wrong password -> a success dialog is displayed after the error message...
The pb is related to the existing `toImportKeys` method of the `MXKEncryptionKeysImportView` (Matrixkit) which calls the onComplete a block when the operation is done (whatever it succeeded or failed).

So unfortunately I would remove this success dialog, because we don't know if the import succeeded or failed.
We may change this onComplete block signature for the future (but we will not update the kit version for the moment at Tchap level)

Reply:
Note: in case of a wrong password, the error message is displayed in English, which is a pity...But again this should be fixed at the matrixKit (I let you check if this is easy to i18n this error , we will then get it later by updating kit version) - This would be a nice to have

Reply:
Alert removed. I'll check for i18n this error translation